frontends/p4/typeChecking/typeChecker.cpp

```diff
@@ -39,8 +39,7 @@ class ConstantTypeSubstitution : public Transform {
                              ReferenceMap* refMap,
                              TypeMap* typeMap) : subst(subst), refMap(refMap), typeMap(typeMap) {
         CHECK_NULL(subst); CHECK_NULL(refMap); CHECK_NULL(typeMap);
-        LOG3("ConstantTypeSubstitution " << subst);
-        setName("ConstantTypeSubstitution"); }
```

Comment:
I use --top4 passSubstring often; I hope that removing the setName won't affect that.

Reply:
It fixes that -- when there are Visitor classes that inherit from each other, we want to use the name of the most-derived class, not the name of the base class.